data/src/gadm/preprocess_gadm.py

```diff
@@ -0,0 +1,156 @@
+
+    return os.path.join(folder, "gadm_410-levels.gpkg")
+
+
+def load_and_clean_gadm_data(gpkg_path):
+    # Load the data
+    print("Loading ADM_0 layer from the geopackage...")
+    adm0_gdf = gpd.read_file(gpkg_path, layer="ADM_0")
+
+    # Clean the geometries
+    print("Cleaning geometries...")
+    adm0_gdf["geometry"] = adm0_gdf["geometry"].apply(fix_geometry)
+
+    # Filter out any empty geometries
+    adm0_gdf = adm0_gdf[~adm0_gdf["geometry"].is_empty]
+    adm0_gdf = adm0_gdf.dropna(subset=["geometry"])
+
+    print("Loaded and cleaned ADM_0 layer with", len(adm0_gdf), "features")
+
+    # Simplify the geometries
+    print("Simplifying geometries...")
+    adm0_gdf = simplify_gdf(adm0_gdf, simplification_tolerance=0.01)
+
+    # Save the cleaned  and simplified data
+    adm0_gdf.to_file("../../raw_data/gadm/gadm_410-levels_ADM_0_cleaned.geojson", driver="GeoJSON")
+    print("Cleaned data saved at: ../../raw_data/gadm/gadm_410-levels_ADM_0_cleaned.geojson")
+
+    return adm0_gdf
+
+
+def fix_geometry(geom):
+    # Validate and fix geometry
+    if not geom.is_valid:
+        geom = make_valid(geom)
+    # Ensure the geometry is in MultiPolygon format
+    if geom.geom_type == "Polygon":
+        geom = MultiPolygon([geom])
+    return geom
+
+
+def simplify_gdf(gdf, simplification_tolerance=0.01, preserve_topology=True):
+    """
+    Simplifies the geometries of a GeoDataFrame in place and returns it.
+
+    Parameters:
+    - gdf (GeoDataFrame): The original GeoDataFrame with geometries to simplify.
+    - simplification_tolerance (float): Tolerance for simplification (e.g., 0.01 for 1% simpl).
+    - preserve_topology (bool): If True, attempts to preserve the topology.
+
+    Returns:
+    - GeoDataFrame: A new GeoDataFrame with simplified geometries.
+    """
+    # Simplify the geometries
+    simplified_gdf = gdf.copy()
+    simplified_gdf["geometry"] = gdf["geometry"].simplify(
+        simplification_tolerance, preserve_topology=preserve_topology
+    )
+    return simplified_gdf
+
+
+# 3. Join geometries with CSV data
+def join_with_csv(geom_gdf, csv_path, key="GID_0"):
+    csv_data = pd.read_csv(csv_path)
+    merged_gdf = geom_gdf.merge(csv_data, left_on=key, right_on="country_code", how="left")
+    # Drop the rows with missing country names
+    merged_gdf = merged_gdf.dropna(subset=["country"])
+    # Save the merged data
+    merged_gdf.to_file("../../raw_data/gadm/gadm_410-levels_ADM_0_merged.geojson", driver="GeoJSON")
+    print("Merged data saved at: ../../raw_data/gadm/gadm_410-levels_ADM_0_merged.geojson")
+
+    return merged_gdf
+
+
+# 4. Generate SQL INSERT statement
+def generate_sql_insert(row):
+    sql_template = """
+    INSERT INTO public.countries
+    (code, "name", continent, region_1, region_2, numeric_code, hdi, geometry)
+    VALUES ('{code}', '{name}', '{continent}'::public.countries_continent_enum, '{region_1}', {region_2}, '{numeric_code}', {hdi}, '{geometry}');
```

Comment:
```suggestion
    VALUES ('{code}', '{name}', '{continent}'::public.countries_continent_enum, '{region_1}', '{region_2}', '{numeric_code}', {hdi}, '{geometry}');
```

The issue is in the lack of quotes for the region_2.  Also, the infamous Cote d'Ivoire might need a special treatment, in my file I have let it as `'Cote d''Ivoire'` so that SQL understands the parsing.

Lastly, the numeric code is parsed as a string with a decimal, such as: `214.0, 36.0, ...`. I guess these should be integers with a length of 3 numbers?